src/mlpack/methods/ann/layer/adaptive_max_pooling_impl.hpp

```diff
@@ -33,6 +33,52 @@ AdaptiveMaxPooling<InputDataType, OutputDataType>::AdaptiveMaxPooling(
   // Nothing to do here.
 }
 
+template<typename InputDataType, typename OutputDataType>
+AdaptiveMaxPooling<InputDataType, OutputDataType>::
+AdaptiveMaxPooling(const AdaptiveMaxPooling& layer) :
+outputWidth(layer.outputWidth),
+outputHeight(layer.outputHeight)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+AdaptiveMaxPooling<InputDataType, OutputDataType>::
+AdaptiveMaxPooling(AdaptiveMaxPooling&& layer) :
+outputWidth(0),
+outputHeight(0)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+AdaptiveMaxPooling<InputDataType, OutputDataType>&
+AdaptiveMaxPooling<InputDataType, OutputDataType>::
+operator=(const AdaptiveMaxPooling& layer)
+{
+  if (this != &layer)
+  {
+    outputWidth = layer.outputWidth;
+    outputHeight = layer.outputHeight;
```

Comment:
Do we need to also copy the `pooling` object?

Reply:
I am sorry but I didn't get you @rcurtin .

Reply:
The `AdaptiveMaxPooling` layer has a member called `pooling`.  Do we need to copy that here too?  I am not sure of the right answer, so that's why I'm asking. :)